uaaextras/webapp.py

```diff
@@ -481,6 +482,37 @@ def signup():
 
         return render_template('error/internal.html'), 500
 
+    @app.route('/redeem-invite', methods=['GET'])
+    def redeem_invite():
+
+        if request.method == 'GET':
+            # Store the validation token to check for UAA invite link
+            verification_code = request.args.get('validation_token')
+
+            # Make sure that the requested URL has validation_token,
+            # otherwise redirect to error page.
+            if 'validation_token' not in request.args:
+                logging.exception('The invitation link was missing a validation token.')
+                return render_template('error/token_validation.html'), 401
+
+            # Check if Redis is working
+            if r:
+                # Check if Redis key was previously requested, otherwise load the UAA invite
+                uaaInviteLink = r.get(verification_code)
+                if uaaInviteLink is None:
+                    logging.info('UAA invite link is expired. {0}'.format(verification_code))
+                    return render_template('error/token_validation.html'), 401
+
+                else:
+                    logging.info('Accessed UAA invite link. {0}'.format(verification_code))
+                    r.delete(verification_code)
```

Comment:
@cnelson This should be good now! I think you might need to "approve" the review.